.github/workflows/codeql-analysis.yml

```diff
@@ -1,71 +1,34 @@
-# For most projects, this workflow file will not need changing; you simply need
-# to commit it to your repository.
-#
-# You may wish to alter this file to override the set of languages analyzed,
-# or to provide custom queries or build logic.
 name: "CodeQL"
 
 on:
   push:
-    branches: [master]
+    branches:
+      - master
+      - "!dependabot/**"
```

Comment:
Seems redundant to say we only want this on `master` and not `dependabot`.
```suggestion
```

Reply:
It's not redundant because we know dependabot always opens a PR.

Reply:
but if we don't want to run code ql on `dependabot` branches why don't we just not include this line? then it will only run code ql on master.

Reply:
Actually, you are right. Not my day today, I need to drink more coffee :)

Reply:
OK now that I think about it, it is needed for PRs. But it's your call. Right now this PR matches the current behavior on master which is to run CodeQL even on dependabot PRs.

Reply:
I don't see any problem with running it on dependabot PRs.